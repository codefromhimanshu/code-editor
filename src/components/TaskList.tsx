// components/TaskList.tsx
import React, { useState }  from 'react';
import TaskDetails from './TaskDetails';

interface TaskListProps {
    tasks: any[];
    onUpdate?: (updatedData: any) => void;
    onDelete?: (taskId: number) => void;
}

const TaskList: React.FC<TaskListProps> = ({ tasks, onUpdate, onDelete }) => {
  const [selectedTask, setSelectedTask] = useState(null);
  // const [filter, setFilter] = useState('All');

  
  // const filteredTasks = tasks.filter(task => filter === 'All' || task.status === filter);

  const getStatusStyles = (status: string) => {
    switch (status) {
      case 'Done':
        return 'bg-green-500';
      case 'In Progress':
        return 'bg-blue-500';
      case 'In Review':
        return 'bg-yellow-500';
      case 'To Do':
        return 'bg-gray-500';
      default:
        return 'bg-gray-300';
    }
  };

  const handleTaskSelect = (task) => {
    setSelectedTask(task);
  };

  const handleTaskUpdate = (updatedTask) => {
    onUpdate(updatedTask)
    setSelectedTask(null);
  };

  const handleTaskDelete = (id) => {
    onDelete(id)
    setSelectedTask(null);
  };


  return (
    <>
     <div className="space-y-4">
      {tasks.map((task) => (
        <div key={task.id} className="flex justify-between items-center bg-white p-4 rounded shadow" onClick={e => handleTaskSelect(task)}>
          {/* <span className={`inline-block w-2 h-2 mr-2 rounded-full ${getStatusStyles(task.status)}`} aria-hidden="true"></span> */}
          <span className={`${task.status === 'Done' ? 'line-through' : ''} flex-grow font-semibold`}>
            {task.title}
          </span>
          <span className={`${task.status === 'Done' ? 'line-through' : ''} flex-grow`}>
            {task.description}
          </span>
          <span className={`text-xs font-semibold mr-2 px-2.5 py-0.5 rounded ${getStatusStyles(task.status)} text-white`}>
            {task.status}
          </span>
        </div>
      ))}
      {
        tasks.length === 0 && (
          <div className='flex items-center"'>
            <span> No Task present</span>
          </div>
          
        )
      }
    </div>
    {selectedTask && (
        <TaskDetails
          task={selectedTask}
          onClose={() => setSelectedTask(null)}
          onUpdate={handleTaskUpdate}
          onDelete={handleTaskDelete}
        />
      )}
    </>
  );
};

export default TaskList;
