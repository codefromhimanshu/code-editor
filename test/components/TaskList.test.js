// tests/TaskList.test.js
import React from "react";
import renderer from "react-test-renderer";
import TaskList from "../../src/components/TaskList";

describe("TaskList", () => {
  const mockTasks = [
    {
      id: "1",
      title: "Test Task 1",
      description: "Description 1",
      status: "To Do",
    },
    {
      id: "2",
      title: "Test Task 2",
      description: "Description 2",
      status: "Done",
    },
  ];

  it("renders correctly with tasks", () => {
    const tree = renderer.create(<TaskList tasks={mockTasks} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders no tasks message when empty", () => {
    const tree = renderer.create(<TaskList tasks={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
