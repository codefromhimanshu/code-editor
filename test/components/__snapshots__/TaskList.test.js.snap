// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList renders correctly with tasks 1`] = `
<div
  className="space-y-4"
>
  <div
    className="flex justify-between items-center bg-white p-4 rounded shadow"
    onClick={[Function]}
  >
    <span
      className=" flex-grow font-semibold"
    >
      Test Task 1
    </span>
    <span
      className=" flex-grow"
    >
      Description 1
    </span>
    <span
      className="text-xs font-semibold mr-2 px-2.5 py-0.5 rounded bg-gray-500 text-white"
    >
      To Do
    </span>
  </div>
  <div
    className="flex justify-between items-center bg-white p-4 rounded shadow"
    onClick={[Function]}
  >
    <span
      className="line-through flex-grow font-semibold"
    >
      Test Task 2
    </span>
    <span
      className="line-through flex-grow"
    >
      Description 2
    </span>
    <span
      className="text-xs font-semibold mr-2 px-2.5 py-0.5 rounded bg-green-500 text-white"
    >
      Done
    </span>
  </div>
</div>
`;

exports[`TaskList renders no tasks message when empty 1`] = `
<div
  className="space-y-4"
>
  <div
    className="flex items-center""
  >
    <span>
       No Task present
    </span>
  </div>
</div>
`;
